### ISSUE TITLE
No network interfaces shown on Pi OS 11 (Debian Bullseye)?

### ISSUE THREAD

**Issue by Jibun-no-Kage** [webmin/webmin#1607]
No network interfaces shown on Pi OS 11 (Debian Bullseye)?  Does WebMin rely on /etc/network/interfaces file?  If so, than any Debian 11 based distribution that has moved to using the dhcpcd service for dhcp IP Address and ethernet interface configuration specifics will not how network configuration in webmin correctly.  Default configuration for most Debian 11 based distributions is now defaulting to dhcpcd service use.  Webmin is blind to this apparently?

**Comment by jcameron** [comment]
Webmin will use /etc/network/interfaces if it exists, or files in /etc/netplan otherwise.

Do you have either of those on Debian 11?

**Comment by Jibun-no-Kage** [comment]
No, not by default.  So webmin shows active interfaces, but no interfaces as configured for startup, which is not true, if dhcpcd is active and that is the default for debian 11 (Pi OS 11 for example).  Debian has been moving to using dhcpcd service as core since 9, and standard in 10, now 11.

**Comment by jcameron** [comment]
So on Debian 11, in which file would you configure a static IP address in? I assume that there's some config in which you specify that you want to use DHCP for an interface?

**Comment by Jibun-no-Kage** [comment]
Look at the DHCPCD service manual information.  You can configure either DHCP or static under DHCPCD, as well as a failback to static if DHCP fails.  I could post examples, but figure you will want to review the detailed information yourself.

**Comment by jcameron** [comment]
DHCPCD controls the assignment of addresses to other machines - what I was wondering is how on Debian 11 you select to use DHCP on an interface. But it sounds like this is set in files under  /etc/network/interfaces.d ?

**Comment by Jibun-no-Kage** [comment]
"DHCP is NOT DHCPCD... note the DIFFERENCE.  DHCPCD is a service that provides reference to DEFINED INTERFACES, which maybe use DHCP based, STATIC or BOTH when a failback configuration is used.  Please DO NOT confuse DHCP with DHCPCD.  This is why I suggested reviewing the documentation of DHCPCD.

Webmin does not show any interfaces configured for startup when 'interfaces' or 'interfaces.d' or 'netplan' is not used for configuration of interfaces at startup.  There another source for network configuration, i.e. DHCPCD,  and so it can be a source for such information.  It is misleading to not show defined (at start) networking information in the webmin UI, because it currently ignores how DHCPCD can quality the start up configuration of interfaces/adapters, no?

**Comment by jcameron** [comment]
Ok, I'm going to have to do some more testing on Debian 11 and get back to you ..

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.

**Comment by iliaross** [comment]
Fully tested and implemented in this PR https://github.com/webmin/webmin/pull/2769 for the Webmin 2.650 release.

I’m really itching to get Home Assistant Smarthomezentrale Raspberry Pi 5 16GB, but it can wait! 😆
